theano/sandbox/cuda/tests/test_opt.py

```diff
@@ -91,6 +92,33 @@ def test_local_gpu_contiguous_gpu_contiguous():
                      if isinstance(node.op, basic_ops.GpuContiguous)])
 
 
+def test_local_assert_no_cpu_op():
+    numpy.random.seed(1)
+    m = numpy.random.uniform(-1, 1, (10, 10)).astype("float32")
+    ms = cuda.shared_constructor(m, name="m_shared")
+    out = theano.tensor.tanh(ms**2 + 3)
+
+    mode_local_assert = mode_with_gpu.including("local_assert_no_cpu_op")
```

Comment:
I would be surprised of that.

On Thu, Apr 30, 2015 at 5:06 PM, Pascal Lamblin <notifications@github.com>
wrote:

> In theano/sandbox/cuda/tests/test_opt.py
> <https://github.com/Theano/Theano/pull/2791#discussion_r29470338>:
>
> > @@ -91,6 +92,33 @@ def test_local_gpu_contiguous_gpu_contiguous():
> >                       if isinstance(node.op, basic_ops.GpuContiguous)])
> >
> >
> > +def test_local_assert_no_cpu_op():
> > +    numpy.random.seed(1)
> > +    m = numpy.random.uniform(-1, 1, (10, 10)).astype("float32")
> > +    ms = cuda.shared_constructor(m, name="m_shared")
> > +    out = theano.tensor.tanh(ms**2 + 3)
> > +
> > +    mode_local_assert = mode_with_gpu.including("local_assert_no_cpu_op")
>
> I'm not sure, but it *might* be possible that building that mode would
> capture the value of config.local_assert_no_cpu_op at the time where the
> mode is built, instead of checking that value each time the optimization is
> applied.
> If that is the case, then you could build mode_local_assert inside each
> of the try: / except: statements.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/2791/files#r29470338>.
>